src/client/transport.js

```diff
@@ -144,77 +141,81 @@ class Transport extends EventEmitter {
         });
     }
 
-    asyncServiceMsg (msg) {
-        return new Promise(resolve => {
-            msg.sessionId = settings.get().sessionId;
+    performRequest (msg, callback) {
```

Comment:
Add comment here, with `TODO` what we need to turn it back to the Promise, once we get rid of `syncServiceMsg`